op-node/rollup/derive/l1_block_info.go

```diff
@@ -506,3 +506,8 @@ func L1InfoDepositBytes(rollupCfg *rollup.Config, sysCfg eth.SystemConfig, seqNu
 func L1InfoInteropLen(dependencySetSize uint8) int {
 	return 4 + 32*5 + 1 + 32*int(dependencySetSize)
 }
+
+// MarshalBinary makes marshalBinaryInterop externally accessible
+func (info *L1BlockInfo) MarshalBinaryInterop() ([]byte, error) {
```

Comment:
This is convenient but doesn't follow the existing API, ie there are no public methods for marshalling to the other types of data. This leads me to believe we don't have the best abstractions. We have 2 options here: 1) call `L1InfoDeposit` and pull the calldata from the tx. This requires a little bit more setup of hardcoding of config. It shouldn't be too difficult. The other option 2) turn the implementation of `marshalBinaryInterop` into an exported function and then use it in the implementation as well as in the fuzzing.

I recommend 1) - its a little more work but it keeps our API cleaner and standardized

Reply:
good idea! implemented 1), but had to _unhardcode_ `InteropDependencySet` in `L1InfoDeposit` since otherwise it wasn't able to include the dependency set as decided by the fuzzer in the diff test. to do this, I introduced a variable in `sysCfg` for `InteropDependencySet`. does it make sense to store `InteropDependencySet` there? alternatively, could store this var in `rollupCfg`. thoughts?

Reply:
This is great, its how it will need to work longer term for sure